internal/pkg/term/selector/selector.go

```diff
@@ -891,6 +913,23 @@ func (s *Select) Environments(prompt, help, app string, finalMsgFunc func(int) p
 	return selectedEnvs, nil
 }
 
+// Connection fetches all the CodeStar in an account and prompts the user to select one.
+func (s *PipelineEnvsConnectionSelect) Connection(msg, help string) (string, error) {
+	connections, err := s.connectionLister.ListConnections()
+	if err != nil {
+		return "", fmt.Errorf("list connections: %w", err)
+	}
+	if len(connections) == 0 {
+		return "", nil
+	}
+	connections = append(connections, "[No thanks]")
+	connection, err := s.prompt.SelectOne(msg, help, connections, prompt.WithFinalMessage(connectionFinalMsg))
+	if err != nil {
+		return "", fmt.Errorf("select CodeStar connection: %w", err)
+	}
+	return connection, nil
```

Comment:
instead of having the client handle the "[No thanks]" response, can we check for it here and return `"", nil` then?